src/async_client.cpp

```diff
@@ -32,7 +32,7 @@ namespace mqtt {
 
 /////////////////////////////////////////////////////////////////////////////
 
-async_client::async_client(const std::string& serverURI, const std::string& clientId)
```

Comment:
Ha. Yes. Agreed.

BTW, I'm not married to this (pulling the string into the `mqtt` namespace). But I have been doing this on a lot of projects lately. It seems to help readability in the library code for such a pervasive type.